### ISSUE TITLE
atom.services is no longer available

### ISSUE THREAD

**Issue by koleesch** [edubkendo/atom-racer#15]
atom.services is no longer available. To register service providers and
consumers, use the `providedServices` and `consumedServices` fields in
your package's package.json.
```
Atom.Object.defineProperty.get (/Applications/Atom.app/Contents/Resources/app/src/atom.js:64:11)
Object.activate (/Users/stefan/.atom/packages/racer/lib/racer.coffee:20:25)
```

**Comment by koleesch** [comment]
I got the message in atom after installing the packages racer and autocomplete-plus

**Comment by koleesch** [comment]
my atom version is 0.177.0

**Comment by alkama** [comment]
Oh, right, the new Atom version does use a new descriptive way to declare services.
As warned in:
- https://github.com/atom-community/autocomplete-plus/wiki/Provider-API#registering-your-provider-with-autocomplete
- https://github.com/atom/atom/pull/5277 

I'll correct this today.

**Comment by joefitzgerald** [comment]
@alkama My understanding is we have a very short window to make this change prior to `atom.services` being removed entirely. Think 1 or 2 atom versions.

The fix is quite simple, and I will update the docs shortly (once I have a proven working example of a provider using this technique). In essence: you include some config in package.json, and add a function that is called in your mainModule.

**Comment by alkama** [comment]
@joefitzgerald yes, that's what I understood too. They sure seem to be in a hurry :) It ain't that dramatic thought.

I understood too that what we need is to add some:
```json
{
  "providedServices": {
    "[NAME]": {
      "description": "lorem ipsum",
      "versions": {
        "[NUMBER]": "[METHOD]"
      }
    }
  }
}
```

What I still dont get is:
- what `[NAME]` should be, if it's the id of our provider, or if it's the string "autocomplete.provider" that we used to use to register via `atom.services.provide()`.
- what `[NUMBER]` should be, if it's the version of our provider, or if it's the version of the autocomplete-plus API that we provide.
- I get that `[METHOD]` is a method we have to create and that becomes the new entrypoint of the provider, but I dont know what it's meant return? should it return the provider object that we used to register with `atom.services.provide()` or some different construct?

My best guess is I would do something like:
```json
{
  "providedServices": {
    "autocomplete.provider": {
      "description": "Intelligent Rust code completion",
      "versions": {
        "1.0.0": "provideAutocompletion"
      }
    }
  }
}
```
and
```coffeescript
module.exports =
  provider: null

  activate: ->
    @provider = new MyProvider()
    return

  provideAutocompletion: ->
    return @provider

  deactivate: ->
    @provider?.dispose()
    @provider = null
    return
```

But I'm probably wrong :)
Anyway, thank for your nice work on autocomplete-plus !

**Comment by joefitzgerald** [comment]
```
{
  "providedServices": {
    "autocomplete.provider": {
      "description": "Intelligent Rust code completion",
      "versions": {
        "1.0.0": "provideAutocompletion"
      }
    }
  }
}
```
is correct.

**Comment by joefitzgerald** [comment]
`return {provider: @provider}` is correct.

**Comment by alkama** [comment]
@joefitzgerald thanks, it indeed seems to work.

**Comment by alkama** [comment]
fixed in v0.9.1
